src/main/java/htsjdk/samtools/reference/FastaReferenceWriter.java

```diff
@@ -0,0 +1,758 @@
+     * @return a reference to this very same writer.
+     * @throws IOException              if such an exception is thrown when actually writing into the output streams/channels.
+     * @throws IllegalArgumentException if either {@code name} or {@code bases} is {@code null} or contains an invalid value (e.g. unsupported bases or sequence names). Also when the
+     *                                  {@code description} contains unsupported characters or {@code basesPerLine} is 0 or negative.
+     * @throws IllegalStateException    if the writer is already closed, a previous sequence (if any was opened) has no base appended to it or a sequence
+     *                                  with such name was already appended to this writer.
+     */
+    public FastaReferenceWriter appendSequence(final String name, final String description, final int basesPerLine, final byte[] bases) throws IOException {
+        return startSequence(name, description, basesPerLine).appendBases(bases);
+    }
+
+    private void assertSequenceOpen() {
+        if (currentSequenceName == null) {
+            throw new IllegalStateException("trying to add bases without starting a sequence");
+        }
+    }
+
+    private void assertIsNotClosed() {
+        if (closed) {
+            throw new IllegalStateException("already closed");
+        }
+    }
+
+    /**
+     * Closes this writer flushing all remaining writing operation input the output resources.
+     * <p>
+     * Further calls to {@link #appendBases} or {@link #startSequence} will result in an exception.
+     * </p>
+     *
+     * @throws IOException if such exception is thrown when closing output writers and output streams.
+     */
+    public void close() throws IOException {
+        if (!closed) {
+            closeSequence();
+            if (sequenceNamesAndSizes.isEmpty()) {
+                throw new IllegalStateException("no sequences where added to the reference");
+            }
+            fastaStream.close();
+            indexWriter.close();
+            dictWriter.close();
+            closed = true;
+        }
+    }
+
+    /**
+     * Convenient method to write a FASTA file with a single sequence.
+     *
+     * @param whereTo     the path to. must not be null.
+     * @param makeIndex   whether the index file should be written at its standard location.
+     * @param makeDict    whether the dictionary file should be written at it standard location.
+     * @param name        the sequence name, cannot contain white space, or control chracter or the header start character.
+     * @param description the sequence description, "" if no description.
+     * @param bases       the sequence bases, cannot be {@code null}.
+     * @throws IOException if such exception is thrown when writing in the output resources.
+     */
+    public static void writeSingleSequenceReference(final Path whereTo, final boolean makeIndex,
```

Comment:
If there is a factory/builder for the writer, the static methods can be moved there

Reply:
Personally I would expect a static method like this to be in the class itself, not a factory method.